### PR TITLE
#168781643  Edit travel requests with "Open" status

### PR DESCRIPTION
**Pull request by sabin18**
#### What does this PR do?
-PR to allow a user to edit request
#### Description of Task to be completed?
- add edit request action
- add edit request functionality
#### How should this be manually tested?
- Clone ft-edit-request-168781643 branch
- run `npm i`
- run `npm run dev`
- Click on edit button to edit your request
#### Any background context you want to provide?
`N/A`
#### What are the relevant pivotal tracker stories?
Screen Shot 2020-01-05 at 03 22 27
Screen Shot 2020-01-05 at 03 22 52

#### Questions:
`N/A`

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]
LGTM

**Comment by ngireric123** [comment]
Great job 
However, @sabin18 kindly raise your coverage to >= 90%

**Review by alainmateso — CHANGES_REQUESTED** [review]
Great job @sabin18 however, I've noticed the following anomalies:
- When editing a multi-city trip, if whatever I edit on one destination is reflected in the rest of the destinations. For example, if I want to edit the reasons for any destination, other destinations change to that as well. In this case, if I edit for example the arrival date, the next trip's information changes to that new date which makes editing that request impossible. 

- In the middle of editing a request, when I press the cancel button, it only disables the fields but the information remains the one I was inputting. This gives the user the wrong impression that their changes were accepted which is not true. I believe canceling should put back the initial information that was present when the request was loaded.

Kindly look into that and kudos for a great job

File: src/views/requests/SingleRequest.js [hunk 1]
```diff
@@ -1,32 +1,65 @@
...
+      isDisabled: true,
+      departureDate: '',
+      returnDate: '',
+      locationId: '',
+      id: '',
+      reasons: '',
+      DepartureDate: '',
+      arrivalDate: '',
+    };
+    this.handleEdit = this.handleEdit.bind(this);
```
Comment (nakiwuge) [inline src/views/requests/SingleRequest.js]: Do not bind use arrow functions and the state directly

File: src/views/requests/SingleRequest.js [hunk 2]
```diff
@@ -1,32 +1,65 @@
...
+    this.setState({
+      isLoading: false,
+      departureDate: this.props.singleData.departureDate,
+      returnDate: this.props.singleData.returnDate,
+      locationId: this.props.singleData.origin.id,
+      id: this.props.singleData.destinations[0].id,
+      reasons: this.props.singleData.destinations[0].reasons,
+      DepartureDate: this.props.singleData.destinations[0].departureDate,
+      arrivalDate: this.props.singleData.destinations[0].arrivalDate,
+      isDisabled: true,
```
Comment (nakiwuge) [inline src/views/requests/SingleRequest.js:47]: destructure `this.props.singleData`

File: src/views/requests/SingleRequest.js [hunk 3]
```diff
@@ -1,32 +1,65 @@
...
+    });
+  }
+
```
Comment (nakiwuge) [inline src/views/requests/SingleRequest.js:51]: Remove this line

File: src/views/requests/SingleRequest.js [hunk 4]
```diff
@@ -35,6 +68,60 @@ export class SingleRequest extends Component {
...
+  handleCancel = (e) => {
+    this.setState({ isDisabled: true });
+  }
+
+  handleChange = (e) => {
+    const { target } = e;
+    const { value } = target;
+    const { name } = target;
```
Comment (nakiwuge) [inline src/views/requests/SingleRequest.js:84]: Add a new line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
Good job @sabin18. Kindly work on the feedback below.
- The user should only view the `Edit request` button when the request is pending
- The manager should not view the `Edit Request` button unless the request belongs to him/her.
- When a user clicks cancel after changing some values the values changed remain. Ensure that when the cancel button is clicked the original data is displayed.
- When the save button is loading disable the cancel button
- Editing the destination field is not working
- Write tests for your feature

**Review by alainmateso — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**